tests/tests.py

```diff
@@ -241,56 +321,42 @@ class TestXML(TestCase):
-        namespace = {'default': 'http://www.w3.org/2005/Atom'}
-        expected_xml_structure = [
-            './default:title',
-            './default:id',
-            './default:updated',
-            './default:author/default:name',
-            './default:content'
-        ]
-        response = self.client.get(reverse('major-event-log:event_atom',
-                                           args=[self.event.id]))
-        atom = ET.fromstring(response.content)
-        for xpath in expected_xml_structure:
-            self.assertIsNotNone(atom.find(xpath, namespace))
+        cls.factory = RequestFactory()
+
+    def get_schema(self, schema_name):
+        """Retrieves the XML schema and returns an XML schema validator."""
+        data_path = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                                 'data')
+        schema_path = os.path.join(data_path, schema_name)
+        with open(schema_path, 'r') as schema_file:
+            schema_root = etree.parse(schema_file)
+        try:
+            schema = etree.XMLSchema(schema_root)
+        except etree.XMLSchemaParseError:
+            assert False
+        return schema
+
+    def test_atom_xml(self):
+        """Validates the Atom items against their schema."""
+        request = self.factory.get('/')
+        response = views.event_atom(request, str(self.event.id))
+        schema = self.get_schema('atom-premis_schema.xsd')
+        atom_item = etree.fromstring(response.content)
+        assert schema.validate(atom_item)
 
     def test_event_premis_xml(self):
-        """Checks for the correct structure within the Atom XML."""
-        namespace = {'prms': 'info:lc/xmlns/premis-v2'}
-        expected_premis_structure = [
-            './prms:eventDetail',
-            './prms:eventOutcomeInformation/prms:eventOutcomeDetail',
-            './prms:eventOutcomeInformation/prms:eventOutcome',
-            './prms:eventType',
-            './prms:linkingAgentIdentifier/prms:linkingAgentIdentifierValue',
-            './prms:linkingAgentIdentifier/prms:linkingAgentIdentifierType',
-            './prms:eventIdentifier/prms:eventIdentifierValue',
-            './prms:eventIdentifier/prms:eventIdentifierType',
-            './prms:eventDateTime'
-        ]
-        response = self.client.get(reverse('major-event-log:event_premis',
-                                           args=[self.event.id]))
-        # Make sure that the PREMIS event has the expected structure.
-        atom = ET.fromstring(response.content)
-        for xpath in expected_premis_structure:
-            self.assertIsNotNone(atom.find(xpath, namespace))
+        """Validates the PREMIS XML against its schema."""
+        request = self.factory.get('/')
+        response = views.event_premis(request, str(self.event.id))
+        schema = self.get_schema('premis_schema.xsd')
+        premis = etree.fromstring(response.content)
+        assert schema.validate(premis)
```

Comment:
We are using the `self.assert*` in some places and a plain `assert` in others. Any explicit reason for this?